src/utils/controls.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Handle a button event and reemits the events.
+ *
+ * @param {string} id - id of the button.
+ * @param {string} evtName - name of the reemitted event
+ * @param {object} component - reference to the component
+ * @param {string} hand - handedness of the controller: left or right.
+ */
+module.exports.onButtonEvent = function (id, evtName, component, hand) {
```

Comment:
Concerned about passing in `component` and having a utility call a bunch of the component's methods.

For example, we did this with `checkControllerPresentAndSetup`, I found it hard to follow. The utils depends on `component.el`, `component.controllerPresent`, `component.addEventListeners`, `component.controllerEventsActive`, `component.injectTrackedControls`. There's a huge amount of interface expected to be available.

Since we want to make available these shared methods and have the controllers implement this common interface, we might want to think early about a higher level API/prototype mixin early. Splitting out to a util makes it seem refactored, but the util and component are heavily coupled and not any more testable. It might be the wrong road to go down.

Reply:
Agree that although the attempt to refactor into utils allowed code reuse, it made it nearly impossible to understand what things were doing without a pretty thorough look

Reply:
You're correct. It's essentially a mixin but passing the `this` as an argument rather than `this.foo()`. I want to stay focused on the task of improving controls. My goal is to easily be able to add `thumbpadup`, `thumbstickup` methods without having to copy code across controllers. There will be just three methods factored out from controllers. In a separate API we can think of a mixin mechanism that consolidates `checkControllerPresentAndSetup` and these factored out methods in a more sustainable pattern.

Reply:
I'm not sure the utils improves the code quality of the controls since it makes things hard to understand, I'd favor copy and pasting code. Ideally, if we're refactoring, we might as well refactor in the right direction. We can merge something temporary if we agree on the mechanism and someone promises to do it soon.

Reply:
Perhaps not agree on the specific code, but that we can have a mixin-style refactor instead soon after you are able to finish the thumbpad/stick work. Then mostly OK with something temporary, these methods are at least less coupled than the tracked-controls utils.